lxd/cluster/recover.go

```diff
@@ -136,6 +136,39 @@ func updateLocalAddress(database *db.Node, address string) error {
 	return nil
 }
 
+// Create patch file for global nodes database.
+func writeGlobalNodesPatch(database *db.Node, nodes []db.RaftNode) error {
+	// No patch needed if there are no nodes
+	if len(nodes) < 1 {
+		return nil
+	}
+
+	var sql strings.Builder
+	for _, node := range nodes {
+		fmt.Fprintf(&sql, "UPDATE nodes SET address = %q WHERE id = %d;\n", node.Address, node.ID)
+	}
+
+	filePath := filepath.Join(database.Dir(), "patch.global.sql")
+	file, err := os.OpenFile(filePath, os.O_APPEND|os.O_CREATE|os.O_WRONLY, 0644)
```

Comment:
Could you just write the sql directly to this file handle rather than using an intermediate strings.Builder?